homeassistant/components/vegehub/__init__.py

```diff
@@ -0,0 +1,139 @@
+"""The Vegetronix VegeHub integration."""
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from http import HTTPStatus
+import logging
+from typing import Any
+
+from aiohttp.hdrs import METH_POST
+from aiohttp.web import Request, Response
+from vegehub import VegeHub, update_data_to_latest_dict
+
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.components.webhook import (
+    async_register as webhook_register,
+    async_unregister as webhook_unregister,
+)
+from homeassistant.const import (
+    CONF_DEVICE,
+    CONF_HOST,
+    CONF_IP_ADDRESS,
+    CONF_MAC,
+    CONF_WEBHOOK_ID,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+
+from .const import DOMAIN, MANUFACTURER, MODEL, NAME, PLATFORMS
+from .coordinator import VegeHubConfigEntry, VegeHubCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class VegeHubData:
+    """Define a data class."""
+
+    coordinator: VegeHubCoordinator
+    hub: VegeHub
+
+
+# The integration is only set up through the UI (config flow)
+async def async_setup_entry(hass: HomeAssistant, entry: VegeHubConfigEntry) -> bool:
+    """Set up VegeHub from a config entry."""
+
+    # Register the device in the device registry
+    device_registry = dr.async_get(hass)
+    device_mac = entry.data[CONF_MAC]
+
+    if not entry.unique_id:
+        raise ConfigEntryError("Error: unable to set up device")
```

Comment:
```suggestion
    assert entry.unique_id:
```
impossible, so just assert for typing

Reply:
If I change this to an assert, the Quality Scale gets mad becaus I'm not raising an error for test-before-setup.

```ERROR] [QUALITY_SCALE] [test-before-setup] Integration does not raise one of {'ConfigEntryAuthFailed', 'ConfigEntryError', 'ConfigEntryNotReady'}```

Should I just do the assert and then add an exemption to the quality_scale.yaml?